flags/builder.go

```diff
@@ -0,0 +1,96 @@
+package flags
+
+import (
+	"sort"
+	"strings"
+)
+
+type ReferenceBuilder struct {
+	max          int // maximum number of flags to find
+	flagsAdded   map[string][]string
+	flagsRemoved map[string][]string
+	foundFlags   map[string]struct{}
+}
+
+func NewReferenceBuilder(max int) *ReferenceBuilder {
+	return &ReferenceBuilder{
+		flagsAdded:   make(map[string][]string),
+		flagsRemoved: make(map[string][]string),
+		foundFlags:   make(map[string]struct{}),
+		max:          max,
+	}
+}
+
+func (b *ReferenceBuilder) MaxReferences() bool {
+	return len(b.foundFlags) >= b.max
+}
+
+func (b *ReferenceBuilder) AddReference(flagKey string, op string, aliases []string) {
+	if op == "+" {
+		b.AddedFlag(flagKey, aliases)
+	} else if op == "-" {
+		b.RemovedFlag(flagKey, aliases)
+	}
+	// ignore
```

Comment:
?

Reply:
i think i'll add a type here so this doesn't happen